agent/operator_endpoint.go

```diff
@@ -265,23 +266,50 @@ func (s *HTTPServer) OperatorAutopilotConfiguration(resp http.ResponseWriter, re
 	}
 }
 
-// FixupConfigDurations is used to handle parsing the duration fields in
-// the Autopilot config struct
-func FixupConfigDurations(raw interface{}) error {
+type durationFixer map[string]struct{}
+
+func NewDurationFixer(fields ...string) durationFixer {
+	d := make(map[string]struct{})
+	for _, field := range fields {
+		d[field] = struct{}{}
+	}
+	return d
+}
+
+// FixupDurations is used to handle parsing any field names in the map to time.Durations
+func (d durationFixer) FixupDurations(raw interface{}) error {
 	rawMap, ok := raw.(map[string]interface{})
 	if !ok {
 		return nil
 	}
 	for key, val := range rawMap {
-		if strings.ToLower(key) == "lastcontactthreshold" ||
-			strings.ToLower(key) == "serverstabilizationtime" {
-			// Convert a string value into an integer
-			if vStr, ok := val.(string); ok {
-				dur, err := time.ParseDuration(vStr)
-				if err != nil {
+		if key == "NodeMeta" {
```

Comment:
It's weird this knows about this field name from a structure elsewhere. Are you worried about clobbering user-supplied keys? I'd make this more explicit where the fixer gets passed names of keys not to explore as well.

Reply:
I forgot to remove this check; it's not needed because the fields to process are explicitly given when the durationFixer is created